sklearn_numba_dpex/common/kernels.py

```diff
@@ -760,14 +739,15 @@ def partial_sum_reduction(
         # a similar memory access pattern than seen at the previous step.
         n_active_sub_groups = n_sub_groups_per_work_group
         for i in range(n_local_iterations):
+
             # At each iteration, half of the remaining work items with the highest id
             # are discarded.
             n_active_sub_groups = n_active_sub_groups // two_as_a_long
             work_item_row_idx = first_row_idx + local_row_idx + n_active_sub_groups
             if (
                 (local_row_idx < n_active_sub_groups) and
-                (col_idx < n_cols) and
-                (work_item_row_idx < sum_axis_size)
+                (_is_cpu or ((col_idx < n_cols) and
+                             (work_item_row_idx < sum_axis_size)))
             ):
```

Comment:
Can you add inline comment to give some precisions regarding this branch?

Reply:
I'll revert this `_is_cpu` change, in fact it does not fix the JIT issue. We must wait for the root cause to be addressed in `numba_dpex`